# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../guppi/gcc4.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur Guppi-0.40.3-orig/src/libguppidata/guppi-data.c Guppi-0.40.3/src/libguppidata/guppi-data.c
--- Guppi-0.40.3-orig/src/libguppidata/guppi-data.c	2001-11-19 05:40:40.000000000 +0000
+++ Guppi-0.40.3/src/libguppidata/guppi-data.c	2006-01-19 22:29:21.545734250 +0000
@@ -1079,8 +1079,6 @@
 static void
 guppi_data_class_init (GuppiDataClass *klass)
 {
-  static void build_menu (GuppiData *, GtkMenu *, gpointer);
-
   GtkObjectClass *object_class = (GtkObjectClass *) klass;
 
   parent_class = gtk_type_class (GTK_TYPE_OBJECT);
diff -Nur Guppi-0.40.3-orig/src/libguppidata/guppi-seq-scalar.c Guppi-0.40.3/src/libguppidata/guppi-seq-scalar.c
--- Guppi-0.40.3-orig/src/libguppidata/guppi-seq-scalar.c	2002-01-14 05:01:17.000000000 +0000
+++ Guppi-0.40.3/src/libguppidata/guppi-seq-scalar.c	2006-01-19 22:30:14.325032750 +0000
@@ -1775,8 +1775,6 @@
 static void
 guppi_seq_scalar_class_init (GuppiSeqScalarClass *klass)
 {
-  static GtkWidget *info_display (GuppiData *);
-
   GtkObjectClass *object_class = (GtkObjectClass *) klass;
   GuppiDataClass *data_class = GUPPI_DATA_CLASS (klass);
   GuppiSeqClass *seq_class = GUPPI_SEQ_CLASS (klass);
diff -Nur Guppi-0.40.3-orig/src/libguppiplot/guppi-axis-markers.c Guppi-0.40.3/src/libguppiplot/guppi-axis-markers.c
--- Guppi-0.40.3-orig/src/libguppiplot/guppi-axis-markers.c	2002-01-21 03:19:05.000000000 +0000
+++ Guppi-0.40.3/src/libguppiplot/guppi-axis-markers.c	2006-01-19 22:32:21.953009000 +0000
@@ -47,10 +47,11 @@
 
 static guint gam_signals[LAST_SIGNAL] = { 0 };
 
+  static void clear (GuppiAxisMarkers *);
+
 static void
 guppi_axis_markers_finalize (GtkObject * obj)
 {
-  static void clear (GuppiAxisMarkers *);
   GuppiAxisMarkers *gal = GUPPI_AXIS_MARKERS (obj);
 
   guppi_finalized (obj);
diff -Nur Guppi-0.40.3-orig/src/libguppiplot/guppi-root-group-item.c Guppi-0.40.3/src/libguppiplot/guppi-root-group-item.c
--- Guppi-0.40.3-orig/src/libguppiplot/guppi-root-group-item.c	2002-01-08 06:31:08.000000000 +0000
+++ Guppi-0.40.3/src/libguppiplot/guppi-root-group-item.c	2006-01-19 22:31:46.090767750 +0000
@@ -153,10 +153,11 @@
   guppi_canvas_item_set_bbox_c (item, 0, 0, xsize, ysize);
 }
 
+  static void canv_size_allocate (GtkWidget *, GtkAllocation *, gpointer);
+
 static void
 changed_scale (GuppiCanvasItem *item, double scale)
 {
-  static void canv_size_allocate (GtkWidget *, GtkAllocation *, gpointer);
   GuppiRootGroupItem *root_item = GUPPI_ROOT_GROUP_ITEM (item);
 
   if (GUPPI_CANVAS_ITEM_CLASS (parent_class)->changed_scale)
diff -Nur Guppi-0.40.3-orig/src/libguppispecfns/mtherr.c Guppi-0.40.3/src/libguppispecfns/mtherr.c
--- Guppi-0.40.3-orig/src/libguppispecfns/mtherr.c	2000-10-20 01:21:48.000000000 +0000
+++ Guppi-0.40.3/src/libguppispecfns/mtherr.c	2006-01-19 22:28:21.645990750 +0000
@@ -58,7 +58,7 @@
 #include <glib.h>
 #include "mconf.h"
 
-static int merror = 0;
+int merror = 0;
 
 /* Notice: the order of appearance of the following
  * messages is bound to the error codes defined
diff -Nur Guppi-0.40.3-orig/src/libguppiuseful/guppi-plug-in.c Guppi-0.40.3/src/libguppiuseful/guppi-plug-in.c
--- Guppi-0.40.3-orig/src/libguppiuseful/guppi-plug-in.c	2001-09-08 05:49:59.000000000 +0000
+++ Guppi-0.40.3/src/libguppiuseful/guppi-plug-in.c	2006-01-19 22:27:56.764435750 +0000
@@ -47,7 +47,7 @@
 
   guppi_finalized (obj);
 
-  guppi_free0 ((gpointer)pi->load_path);
+  guppi_free0 (pi->load_path);
 
   if (parent_class->finalize)
     parent_class->finalize (obj);
